software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -155,29 +155,42 @@ public void attachLifecycleEffectors(Entity entity) {
                 .build();
     }
 
+    /** @see {@link #newStartEffector()} */
+    public Effector<Void> newSuspendEffector() {
+        return Effectors.effector(Void.class, "suspend")
+                .description("Suspend the process/service represented by an entity")
+                .parameter(StopSoftwareParameters.STOP_PROCESS_MODE)
+                .parameter(StopSoftwareParameters.STOP_MACHINE_MODE)
+                .impl(newSuspendEffectorTask())
+                .build();
+    }
+
     /**
-     * Returns the {@link TaskFactory} which supplies the implementation for the start effector.
+     * Returns the {@link EffectorBody} which supplies the implementation for the start effector.
      * <p>
      * Calls {@link #start(Collection)} in this class.
      */
     public EffectorBody<Void> newStartEffectorTask() {
-        return new EffectorBody<Void>() {
-            @Override
-            public Void call(ConfigBag parameters) {
-                Collection<? extends Location> locations  = null;
+        return new StartEffectorBody();
```

Comment:
@sjcorbett @neykov This is a big scary. I just hit it in my testing for a customer release. But that's because I tested a release candidate a day ago and then again today with this change.

Looking at what is persisted (with non-static classes etc), it feels scary for trying to support (or transform) that in future releases.

I'm going to change SoftwareProcess. LIFECYCLE_EFFECTOR_TASKS so that if the config is null it defaults to as before; if it's non-null then it will use the config value. That means we can take our time over whether we want to have a SoftwareProcessDriverLifecycleEffectorTasks instances persisted by pretty much every customer.